src/Orleans/Async/Cancellation/ICancellationSourcesExtension.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading.Tasks;
+using Orleans.Concurrency;
+using Orleans.Runtime;
+
+namespace Orleans.Async
+{
+    internal interface ICancellationSourcesExtension : IGrainExtension, IGrain
+    {
+        [AlwaysInterleave]
+        Task CancelTokenSource(GrainCancellationToken token);
```

Comment:
If users cancel before actual send - that's not necessary a bug in his code, as he may have recieved this token from outside ( e.g from another grain)  thus he cann't make any assuptions about when the token will get cancelled. Anyway in such case already cancelled token will be sent and work fine. 
>If he send them in the right order but did not first await the first call
Usually operations that requires cancellation tokens is taking relatively considerable amount of time, thats why they need tokens. But if user will await such operation before cancelling - there will be no point in cancelling, right? And that's why the cancellation extension `Cancel` method have `AlwaysInterleave` attribute - to be able to interrupt executing call.